### PR TITLE
[xcm-builder] Replaced deprecated CurrencyAdapter with FungibleAdapter

### PR DESCRIPTION
**Pull request by bkontur**
I found out during the cleanup of this deprecation message in the `polkadot-fellows` repository that we deprecated `CurrencyAdapter` without making the recommended changes.

## TODO
- [ ] fix `polkadot-fellows` bump to 1.6.0 https://github.com/polkadot-fellows/runtimes/pull/159

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5179827

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5180409

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5180408

**Comment by bkontur** [comment]
@franciscoaguirre I see that tests/benchmarks fails, also I am checking your PR: https://github.com/paritytech/polkadot-sdk/pull/2684
> and I'll change it to the new FungibleAdapter in a following PR.
> The two stages are separated so as to not bloat this PR with some name fixes in tests.

What about that following PR? :)

also I changed CurencyAdapter to FungibleAdapter for fellows and tests started to fail: https://github.com/polkadot-fellows/runtimes/actions/runs/7869327889/job/21468225062?pr=159

**Comment by franciscoaguirre** [comment]
Pallet balances' `fungible` implementation emits different events than its `Currency` implementation. I had to change them in the tests to make them pass

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5190971

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5190972

**Comment by franciscoaguirre** [comment]
Need to change all the other runtimes

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5195504

**Comment by franciscoaguirre** [comment]
Seems there's a problem with benchmarks... Looking into it

File: bridges/snowbridge/runtime/test-common/src/lib.rs [hunk 1]
```diff
@@ -188,6 +188,7 @@ pub fn send_transfer_token_message_success<Runtime, XcmConfig>(
 			let channel_id: ChannelId = origin.into();
 
 			let nonce = snowbridge_pallet_outbound_queue::Nonce::<Runtime>::try_get(channel_id);
+			dbg!(&nonce);
```
Comment (KiChjang) [inline bridges/snowbridge/runtime/test-common/src/lib.rs:191]: Should probably remove this.

File: polkadot/xcm/pallet-xcm-benchmarks/src/fungible/benchmarking.rs [hunk 2]
```diff
@@ -79,7 +82,7 @@ benchmarks_instance_pallet! {
 	}: {
 		executor.bench_process(xcm)?;
 	} verify {
-		assert!(T::TransactAsset::balance(&sender_account).is_zero());
+		assert!(T::TransactAsset::balance(&sender_account) < sender_account_balance_before);
```
Comment (seadanda) [inline polkadot/xcm/pallet-xcm-benchmarks/src/fungible/benchmarking.rs:85]: Is this a tight enough bound on this test? It's just checking that it decreases by any amount, whereas do we not expect it to change by some predictable amount?

**Review by seadanda — APPROVED** [review]

File: bridges/snowbridge/runtime/test-common/src/lib.rs (hunk 1 above)
Comment (bkontur) [reply bridges/snowbridge/runtime/test-common/src/lib.rs:191]: cc @claravanstaden

File: polkadot/xcm/pallet-xcm-benchmarks/src/fungible/benchmarking.rs (hunk 2 above)
Comment (franciscoaguirre) [reply polkadot/xcm/pallet-xcm-benchmarks/src/fungible/benchmarking.rs:85]: I agree it should be tighter. Will look into it on another PR 😄

**Pull request merged**